lib/client.js

```diff
@@ -435,12 +439,23 @@ class Parser extends HTTPParser {
       return true
     }
 
+    if (headers['keep-alive']) {
+      const m = headers['keep-alive'].match(/timeout=(\d+)/)
+      if (m) {
+        const keepAliveTimeout = parseInt(m[1]) * 1000
+        // Set timeout to half of hint to account for timing inaccuracies.
+        client[kKeepAliveTimeout] = Math.min(keepAliveTimeout / 2, client[kIdleTimeout])
```

Comment:
`/ 2` seems to be too much. Why not `keepAliveTimeout - 100`? That should be pretty accurate even on slow machines.